src/data/polynomial/div.lean

```diff
@@ -430,6 +431,17 @@ end
 
 variable {R}
 
+lemma eval_ring_hom_ker (x : R) : (eval_ring_hom x).ker = ideal.span {X - C x} :=
+by { ext y, simpa only [ideal.mem_span_singleton, dvd_iff_is_root] }
+
+/-- For a commutative ring `R`, evaluation at an element `x : R` induces an `R`-algebra isomorphism
+$R[X] / \langle X - x \rangle \cong R$. -/
+noncomputable def quotient_span_X_sub_C_alg_equiv (x : R) :
```

Comment:
Since this is a `def`, can you add a lemma stating what the forward and reverse directions are?

Does `@[simps]` generate anything reasonable in either direction?

Reply:
`@[simps]` generates `_apply` and `_symm_apply` - I guess we don't need to manually write the forward/reverse directions

Reply:
What are the statements of those lemmas though? Are they useful?

Reply:
Hmm yeah, they don't seem to be very useful - just unwrapping the definition (and I have a feeling the linter might time out). I think this is also the case for `ideal.quotient_equiv_alg_of_eq`, `ideal.quotient_equiv_alg`, `ideal.quotient_equiv`, etc so `@[simps]` were never added for those. I don't see a scenario where we would apply this definition on elements though - I see it more as an abstract isomorphism.

Reply:
The `@[simps]` caused a lemma (unrelated to this?) to time out, so I removed it

Reply:
I think you still should have a lemma here, even if it's manual. The forward lemma would probably be the lean version of "evaluation at an element" which currently is only in the docstring.

I guess the reverse direction is just `algebra_map`?

Reply:
I need the `equiv` to be computable to prove the reverse direction, and it seems like `eval_surjective` is no longer necessary

Reply:
Excellent, I hadn't even thought about that. You certainly could have proved it anyway, but this is nicer than what I had in mind